mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx

```diff
@@ -0,0 +1,108 @@
+import { View } from '@components'
+import { IconName, IconType, ThemedIcon, ThemedTextV2, ThemedTouchableOpacityV2 } from '@components/themed'
+import { useFeatureFlagContext } from '@contexts/FeatureFlagContext'
+import { NavigationProp, useNavigation } from '@react-navigation/native'
+import { RootState } from '@store'
+import { tailwind } from '@tailwind'
+import { translate } from '@translations'
+import { ScrollView, Text } from 'react-native'
+import { useSelector } from 'react-redux'
+import { PortfolioParamList } from '../PortfolioNavigator'
+
+export interface ActionButtonsProps {
+  name: string
+  icon: IconName
+  iconType: IconType
+  onPress: () => void
+  badge?: string | number
+}
+
+export function ActionButtons (): JSX.Element {
+  const { isFeatureAvailable } = useFeatureFlagContext()
+  const navigation = useNavigation<NavigationProp<PortfolioParamList>>()
+  const { futureSwaps } = useSelector((state: RootState) => state.futureSwaps)
+
+  const actions: ActionButtonsProps[] = [
+    {
+      name: translate('components/ActionButtons', 'Send'),
+      icon: 'arrow-up-right',
+      iconType: 'Feather',
+      onPress: () => navigation.navigate('Send')
+    }, {
+      name: translate('components/ActionButtons', 'Receive'),
+      icon: 'arrow-down-left',
+      iconType: 'Feather',
+      onPress: () => navigation.navigate('Receive')
+    }, {
+      name: translate('components/ActionButtons', 'Swap'),
+      icon: 'repeat',
+      iconType: 'Feather',
+      onPress: () => navigation.navigate({
+        name: 'Convert',
+        params: { mode: 'utxosToAccount' },
+        merge: true
+      })
+    }, {
+      name: translate('components/ActionButtons', 'Transactions'),
+      icon: 'calendar',
+      iconType: 'Feather',
+      onPress: () => navigation.navigate('TransactionsScreen')
+    }
+  ]
+
+  if (isFeatureAvailable('future_swap') && futureSwaps.length > 0) {
+    actions.splice(2, 0, {
+      name: translate('components/ActionButtons', 'Future swap'),
+      icon: 'clock',
+      iconType: 'Feather',
+      badge: futureSwaps.length > 9 ? '9+' : futureSwaps.length + 1,
```

Comment:
badge: futureSwaps.length > 9 ? '9+' : futureSwaps.length,
```suggestion
      badge: futureSwaps.length > 9 ? '9+' : futureSwaps.length,
```

We aren't displaying the count of future swaps before. Can help ask if the count should combine if the source and destination token is the same?

For example, for this future swap, is this counted as `1` or `2`?
<img width="365" alt="image" src="https://user-images.githubusercontent.com/13292662/181679179-8f8e7f47-98f6-486a-b2bc-70ebe1a0bbde.png">